src/RestrictedContent/index.tsx

```diff
@@ -0,0 +1,70 @@
+import type { Poool } from 'poool-access';
+import {
+  type ComponentPropsWithoutRef,
+  type MutableRefObject,
+  type ReactNode,
+  Children,
+  cloneElement,
+  forwardRef,
+  useRef,
+  useImperativeHandle,
+  ReactElement,
+} from 'react';
+
+export declare interface RestrictedContentProps
+  extends ComponentPropsWithoutRef<any> {
+  /**
+   * Way to hide content see
+   * [Access configuration]
+   * (https://www.poool.dev/docs/access/javascript/access/configuration#mode)
+   * for more informations.
+   *
+   * More infos:
+   * https://www.poool.dev/docs/access/javascript/access/configuration
+   */
+  mode?: Poool.AccessConfigOptions['mode'];
```

Comment:
same here, I think it could be great to extends AccessConfigOptions and pick interesting types